### DISCUSSION TITLE
Texture transform parameters (offset, rotation and scale)

### DISCUSSION THREAD

**Discussion by fbenedetti06** [google/model-viewer#3226] [Q&A]
Hi all,

I'm a very beginner with model-viewer and 3D model. So I apologize by advance if my question is stupid!

With Blender I made a simple 3D model.
I used an external image as a texture and I changed offset, rotation and scale parameters to repeated it all over the surface (the render look like a tiled wall).

With model-viewer I would like to replace programmatically initial texture image by another.
Unfortunatly it seems that all transform parameters are lost (now the render look like a single one big tile all over the wall).
I can't find a way to restore these parameters by program nor a function to set these values.

Here the part of Gltf 3D file I identified to be desired settings to resize and repeat the tiles (generated by Blender)
![image](https://user-images.githubusercontent.com/49648330/155746400-75c87d93-48f1-4545-b802-dc643364cd29.png)

Here lines of JS code I use to change programmatically the image:
![image](https://user-images.githubusercontent.com/49648330/155748163-b04ec732-ff79-43d3-8a41-58a92f581378.png)

Any idea?

Fred

**Comment by samaneh-kazemi** [comment]
Hi @fbenedetti06, for reaching out. Do you mind attaching your 3D model and a link to your project so we can debug it easier? You can use glitch for that.

**Comment by fbenedetti06** [comment]
Thank you for such a quick response.

I simplified to the max this code and you can run it here: https://www.my-net.fr/github_test/test.htm
There is no index so you can access to all files: https://www.my-net.fr/github_test/

Also here attached: [test.zip](https://github.com/google/model-viewer/files/8145262/test.zip)

Fred

**Comment by fbenedetti06** [comment]
After more investigation it seems that sophisticated textures use an extension called "KHR_texture_transform".
This common and official extension add to basic gltf file format some offset, scalling and rotation properties to alter texture images.

I constat that model-viewer is able to provide a correct render when gltf file use this extension.
The issue is model-viewer do not provide functions (nor extra arguments for existing functions) to read/write these properties.

Is there a way to change only texture image WITHOUT change transformation properties?
Is there a way to patch directly 3D model in memory to change some data?

**Reply by elalish** [reply]
Yes, this looks like a bug, which is now here: https://github.com/google/model-viewer/issues/3240

Thanks for the report!

**Reply by elalish** [reply]
I think I've fixed this now in #3263.
